cmd/rosetta-dispatcher-server/main.go

```diff
@@ -0,0 +1,297 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	url2 "net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"time"
+
+	"github.com/labstack/echo/v4"
+	"github.com/labstack/echo/v4/middleware"
+	"github.com/optakt/flow-dps/rosetta/identifier"
+	"github.com/rs/zerolog"
+	"github.com/spf13/pflag"
+	"github.com/ziflex/lecho/v2"
+)
+
+const (
+	success = 0
+	failure = 1
+)
+
+type Spork struct {
+	First      uint64
+	Last       uint64
```

Comment:
I don't think we need a first and a last for each spork, having just the root height of each spork should be enough. What do you think?

Reply:
I think there is value in making it explicit. Makes checks easier and makes mistakes more obvious.

Reply:
I don't feel like it's so bad with just the root heights, wdyt?

```go
func (s sporks) ProxyTarget(height uint64) *middleware.ProxyTarget {
	// Iterate over all sporks except the last one and check if the requested height is between the root height of this
	// element and the next's.
	for i := range s[:len(s)-1] {
		if height >= s[i].RootHeight && height <= s[i+1].RootHeight {
			return s[i].Target.ToProxyTarget()
		}
	}

	// If no match was found, it means the height is above the root height of the last listed spork. Return its target.
	return s[len(s)-1].Target.ToProxyTarget()
}
```

The previous implementation was the following:

```go
func (s SporkList) ServerForHeight(height uint64) (*middleware.ProxyTarget, error) {
	// Assume sorted and more than zero items
	first := s[0].First
	last := s[len(s)-1].Last

	for _, spork := range s {
		if height >= spork.First && height <= spork.Last {
			return spork.ProxyTaget, nil
		}
	}

	return nil, fmt.Errorf("spork for height %d not found", height)
}
```